addons/pos_online_payment/models/pos_payment_method.py

```diff
@@ -97,7 +97,12 @@ def _force_online_payment_values(vals, if_present=False):
             vals['payment_method_type'] = 'none'
 
     def _get_payment_terminal_selection(self):
-        return super(PosPaymentMethod, self)._get_payment_terminal_selection() if not self.is_online_payment else []
+        selection = []
+        if len(self.ids) == 0:
+            return super()._get_payment_terminal_selection() if not self.is_online_payment else []
+        for method in self:
+            selection.append(super(PosPaymentMethod, method)._get_payment_terminal_selection() if not method.is_online_payment else [])
+        return selection
```

Comment:
```suggestion
        terminal_selections = []
        for method in self:
            if not method.is_online_payment:
                terminal_selections.append(super(PosPaymentMethod, self)._get_payment_terminal_selection())
        else:
            return super()._get_payment_terminal_selection() if not self.is_online_payment else []    
        return terminal_selections
```

I would do something like this